pennylane/beta/tapes/tape.py

```diff
@@ -1337,3 +1660,75 @@ def jacobian(self, device, params=None, **options):
             jac[:, idx] = g.flatten()
 
         return jac
+
+    def iterator(self):
+        """Hack to make iterating over the full queue easier."""
+        queue = self.operations + self.measurements
+        for o in queue:
+            yield o
+
+
+class BatchTape(QuantumTape):
+    """A quantum tape recorder whose queue records elements which are interpreted as alternatives to each other.
+
+    A QuantumTape containing BatchTape operations is interpreted as, and can be separated into,
+    a batch of quantum tapes.
+
+     **Example**
+
+    Alternative pathways through the circuit can be recorded using a ``BatchTape``:
+
+    .. code-block:: python
+
+        with QuantumTape() as tape:
+
+            with QuantumTape() as tape2:
+                qml.Rot(0.543, 0.1, 0.4, wires=0)
+
+            qml.CNOT(wires=['a', 0])
+
+            with BatchTape() as batch:
+                qml.RY(0.2, wires='a')
+                qml.RX(0.2, wires='a')
+
+            probs(wires=0), probs(wires='a')
+
+    The ``tape`` now contains a ``BatchTape`` object:
+
+    >>> tape.operations
+    [BasisState(array([1, 1]), wires=[0, 'a']),
+     CNOT(wires=[0, 'a']),
+     <BatchTape: wires=[0], n_batches=2>]
+
+    We can also batch measurements:
+
+        .. code-block:: python
+
+        with QuantumTape() as tape:
+
+            qml.CNOT(wires=['a', 0])
+
+            with BatchTape() as batch:
+                probs(wires=0)
+                probs(wires='a')
+
+
+    The ``tape`` contains a ``BatchTape`` object:
+
+    >>> tape.measurements
+    [<BatchTape: wires=[0, 'a'], n_batches=2>]
+    """
+
+    def __init__(self, name=None):
+        super().__init__(name=name)
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__}: wires={self.wires.tolist()} " \
+               f"batches={self.n_batches}, name={self.name}>"
+
+    @property
+    def n_batches(self):
+        return len(self.operations) + len(self.observables)
+
+    def analytic_pd(self, idx, device, params=None, **options):
```

Comment:
Needed to implement this function, it is a dummy here. This would be solved by a `BaseQuantumTape` class from which I could inherit.